3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/AnonIpEnrichment.scala

```diff
@@ -118,23 +167,60 @@ case class AnonIpEnrichment(
    * anonymizeIp("94.15.223.151", Three)
    * => "94.x.x.x"
    *
+   *
+   * For IPv6 either the form defined in RFC 2732
+   *  or the literal IPv6 address format defined in RFC 2373 is accepted
+   *
    * TODO: potentially update this to return
    * a Validation error or a null if the IP
    * address is somehow invalid or incomplete.
    *
    * @param ip The IP address to anonymize
    * @return the anonymized IP address
    */
-  import AnonOctets._
-  def anonymizeIp(ip: String): String =
-    Option(ip)
-      .map(
-        _.split("\\.").zipWithIndex
-          .map {
-            case (q, i) => {
-              if (octets.id >= All.id - i) "x" else q
-            }
-          }
-          .mkString("."))
-      .orNull
+  val IPv4MappedAddressPrefix = "::FFFF:"
+  val MaskChar                = "x"
+
+  def anonymizeIp(ipOrNull: String): String =
+    Option(ipOrNull).map { ip =>
+      Try(InetAddress.getByName(ip))
+        .map {
+          case _: Inet4Address => anonymizeIpV4(ip)
+          case _: Inet6Address => anonymizeIpV6(ip)
+        }
+        .getOrElse(tryAnonymizingInvalidIp(ip))
+    }.orNull
+
+  private def anonymizeIpV4(ip: String): String = {
+    def masked(ipv4: String) = {
+      val splitted = ipv4.split("\\.")
+      splitted
+        .take(AnonIPv4Octets.All.id - ipv4Octets.id)
+        .toList
+        .padTo(splitted.size, MaskChar)
+        .mkString(".")
+    }
+
+    if (ip.startsWith(IPv4MappedAddressPrefix))
+      IPv4MappedAddressPrefix + masked(ip.replace(IPv4MappedAddressPrefix, ""))
+    else masked(ip)
+  }
+
+  private def anonymizeIpV6(ip: String): String =
+    InetAddress
+      .getByName(ip)
+      .getHostAddress
```

Comment:
`InetAddress.getByName(ip).getHostAddress` when provided with `ip` (not `host name`) does not do any dns lookup calls. It's used here to not have to deal with tricky IPv6 format, which allows omitting certain parts. `hetHostName` returns always 8 segments, so it makes it easy to anonymize.